spec/lib/puppet_spec/files.rb

```diff
@@ -43,6 +43,20 @@ def self.file_containing(name, contents)
     file
   end
 
+  def script_containing(name, contents) PuppetSpec::Files.script_containing(name, contents) end
+  def self.script_containing(name, contents)
+    file = tmpfile(name)
+    if Puppet.features.microsoft_windows?
+      file += '.bat'
+      text = contents[:windows]
+    else
+      text = contents[:posix]
+    end
+    File.open(file, 'w') { |f| f.write(text) }
```

Comment:
Batch scripts seem to be ok with just "\n", so defaulting to 'wb' would probably be safe.